go/base/context.go

```diff
@@ -736,3 +738,12 @@ func (this *MigrationContext) ReadConfigFile() error {
 
 	return nil
 }
+
+// DisableBinLogForQuery if UseBinLog is false will wrap the query by wrapping
+// it in SET sql_log_bin = OFF/ON
+func (this *MigrationContext) DisableBinLogForQuery(query string) string {
+	if this.UseBinLog {
+		return query
+	}
+	return fmt.Sprintf("SET sql_log_bin = OFF; %s; SET sql_log_bin = ON;", query)
```

Comment:
I realize this code may be going away, but just a general word of caution with this multiStatements=true trick: wrapping the query will break the ability of DML queries to return results about # of rows affected or last-insert-id. I'm also not sure what happens if the query errors -- you may need to be very careful to avoid having some random connection returned to your connection pool with sql_log_bin=0 still in place...

The idiomatic way to set session variables in database/sql is via [DB.Conn()](https://golang.org/pkg/database/sql/#DB.Conn), which allows you to operate on a single session across multiple queries.

Reply:
> The idiomatic way to set session variables in database/sql is via DB.Conn(), which allows you to operate on a single session across multiple queries.

Yeah the issue was that we didn't want all statements to have `sql_log_bin = OFF`. This was a way for me to opt-in on the statement level.

However, I wonder if it would have been better handled like the `singletonDB` to direct statements with/without binary logging to a particular connection (or connection pool).

Either way no biggy. @shlomi-noach suggested to rip out this part of the PR for now and I'm good with that.